qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/DateFieldsIT.java

```diff
@@ -0,0 +1,195 @@
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.upgrades;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.Response;
+import org.junit.BeforeClass;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * This is test is meant to verify that when upgrading from 6.x version to 7.7 or newer it is able to parse date fields with joda pattern.
+ * This test cannot be implemented in yml because in mixed cluster
+ * there are 3 options of warnings to be returned (it was refactored few times).
+ * A special flag on DocValues is used to indicate that an index was created in 6.x and has a joda pattern.
+ * When upgrading from 7.0-7.6 to 7.7 there is no way to tell if a pattern was created in 6.x as this flag cannot be added.
+ * @see org.elasticsearch.search.DocValueFormat.DateTime
+ */
+public class DateFieldsIT extends AbstractRollingTestCase {
+    private static final Version UPGRADE_FROM_VERSION =
+        Version.fromString(System.getProperty("tests.upgrade_from_version"));
+
+    private static final String V_6_8_1_PLUS_WARNING = "'Y' year-of-era should be replaced with 'y'. Use 'Y' for week-based-year.; " +
+        "'Z' time zone offset/id fails when parsing 'Z' for Zulu timezone. Consider using 'X'." +
+        " Prefix your date format with '8' to use the new specifier.";
+
+    private static final String V_7_0_0_PLUS_WARNING = "'Y' year-of-era should be replaced with 'y'. Use 'Y' for week-based-year.; " +
+        "'Z' time zone offset/id fails when parsing 'Z' for Zulu timezone. Consider using 'X'. " +
+        "Use new java.time date format specifiers.";
+
+    private static final String V_6_8_0_WARNING = "Use of 'Y' (year-of-era) will change to 'y' in the next major version of Elasticsearch. "
+        + "Prefix your date format with '8' to use the new specifier.";
+
+    @BeforeClass
+    public static void init(){
+        assumeTrue("upgrading from 7.0-7.6 will fail parsing joda formats",
+            UPGRADE_FROM_VERSION.before(Version.V_7_0_0));
+    }
+
+    public void testJodaBackedDocValueAndDateFields() throws Exception {
+
+        switch (CLUSTER_TYPE) {
+            case OLD:
+                Request createTestIndex = indexWithDateField("joda_it", "YYYY-MM-dd'T'HH:mm:ssZZ");
```

Comment:
nit: it's unclear to me what the suffix "_it" means. perhaps use the index names "joda_time" and "java_time"?

Reply:
Initially I had it named `joda` and `java` but I was clashing with another yml test which  was using the same index names.
Will rename this to joda_time/java_time